mlrun/datastore/targets.py

```diff
@@ -388,6 +458,30 @@ def add_writer_state(
             features=features, timestamp_key=timestamp_key, key_columns=None
         )
 
+        partition_cols = None
+        if self.key_bucketing_number is not None:
+            partition_cols = [
+                "$key"
+                if self.key_bucketing_number == 0
+                else ("$key", self.key_bucketing_number)
+            ]
+        if self.partition_cols:
+            partition_cols = partition_cols or []
```

Comment:
Why not init partition_cols as [] instead of None and skip this line?

Reply:
It needs to be `None` when partitioning is off.